includes/class-wc-payments-customer-service.php

```diff
@@ -165,22 +165,21 @@ public function create_customer_for_user( WP_User $user, array $customer_data ):
 	 *
 	 * @param int      $user_id ID of the WP user to associate with the customer.
 	 * @param WC_Order $order   Woo Order.
+	 *
 	 * @return string           WooPayments customer ID.
-	 */
+	 * @throws API_Exception    Throws when server API request fails.
+*/
 	public function get_or_create_customer_id_from_order( int $user_id, WC_Order $order ): string {
 		// Determine the customer making the payment, create one if we don't have one already.
-		$customer_id = $this->get_customer_id_by_user_id( $user_id );
+		$customer_id   = $this->get_customer_id_by_user_id( $user_id );
+		$customer_data = self::map_customer_data( $order, new WC_Customer( $user_id ) );
+		$user          = get_user_by( 'id', $user_id );
```

Comment:
Seems like this doesn't work when creating customers in incognito (w/o user logged in) because `create_customer_for_user()` requires a `WP_User` object, and this line returns `false` when `$user_id` is 0 (which it is when no customer is logged in).

The current implementation in the gateway is slightly weird: It's either `$order->get_user()` (🆗) or `wp_get_current_user()` (⛔) otherwise. Then we're working with a dummy `WP_User` object, which requires the weird checks here: https://github.com/Automattic/woocommerce-payments/blob/e3b86b4e0bd9fe9311dcc801d9b79c5f709f9acc/includes/class-wc-payments-customer-service.php#L151-L153C4

I'm marking the PR as blocked for now, let's discuss how to solve this tomorrow.